src/dataworker/PoolRebalanceUtils.ts

```diff
@@ -594,44 +594,48 @@ export function generateMarkdownForRootBundle(
   });
 
   let slowRelayLeavesPretty = "";
-  slowRelayLeaves.forEach((leaf, index) => {
+  slowRelayLeaves.forEach((leaf: interfaces.SlowFillLeaf, index) => {
     const outputToken = sdkUtils.getRelayDataOutputToken(leaf.relayData);
-    const outputTokenDecimals = hubPoolClient.getTokenInfo(leaf.relayData.destinationChainId, outputToken).decimals;
+    const destinationChainId = sdkUtils.getSlowFillLeafChainId(leaf);
+    const outputTokenDecimals = hubPoolClient.getTokenInfo(destinationChainId, outputToken).decimals;
+    const lpFeePct = sdkUtils.getSlowFillLeafLpFeePct(leaf);
 
     const slowFill: Record<string, string> = {
       // Shorten select keys for ease of reading from Slack.
       depositor: shortenHexString(leaf.relayData.depositor),
       recipient: shortenHexString(leaf.relayData.recipient),
-      originChainId: leaf.relayData.originChainId,
-      destinationChainId: leaf.relayData.destinationChainId,
-      depositId: leaf.relayData.depositId,
+      originChainId: leaf.relayData.originChainId.toString(),
+      destinationChainId: destinationChainId.toString(),
+      depositId: leaf.relayData.depositId.toString(),
       message: leaf.relayData.message,
       // Fee decimals is always 18. 1e18 = 100% so 1e16 = 1%.
-      realizedLpFeePct: `${formatFeePct(leaf.relayData.realizedLpFeePct)}%`,
+      realizedLpFeePct: `${formatFeePct(lpFeePct)}%`,
     };
 
     if (sdkUtils.isV2SlowFillLeaf(leaf)) {
```

Comment:
I think the `isV2SlowFillLeaf()` predicate requires additional hinting here, otherwise it doesn't work as expected and tsc has problems narrowing the type. RE the following change I made on my end, as well as this comment:
- https://github.com/across-protocol/relayer-v2/pull/1184/commits/0dbf0ee42a093b8105b74ab192e7969b9be20296
- https://github.com/across-protocol/relayer-v2/pull/1155/files/28581ec888f50bf51da19561ebfbaf8c3e87c1ea#r1491916216

```suggestion
    if (sdkUtils.isV2SlowFillLeaf<V2SlowFillLeaf, V3SlowFillLeaf>(leaf)) {
```